### PR TITLE
[Pointcloud] Fixes in Pointcloud Gem

### PR DESCRIPTION
**Pull request by zakmat**
This PR applies several fixes into Pointcloud Gem:
* rectifies a regression where PointcloudConfigurationBus was not enabled for Editor Component
* Removes log flooding with failed re-enable of feature processor by querying before enable attempt
* fixes bounds calculation for invalid pointclouds - applying transform was making them valid but incorrect.

The PR also cleans up headers a little and add some comments

### CODE REVIEW

**Review by norbertprokopiuk — COMMENTED** [review]
Most of included changes are fine but i have some doubts about local bounds checking. current solution makes it impossible to select point cloud if point (0,0,0) is outside of camera frustum

File: Gems/Pointcloud/Code/Source/Tools/Components/PointcloudEditorComponent.cpp [hunk 1]
```diff
@@ -80,7 +78,10 @@ namespace Pointcloud
         AZ::Transform transform = AZ::Transform::CreateIdentity();
         AZ::TransformBus::EventResult(transform, GetEntityId(), &AZ::TransformBus::Events::GetLocalTM);
         AZ::Aabb bounds = m_controller.GetBounds();
-        bounds.ApplyTransform(transform);
+        if (bounds.IsValid())
+        {
+            bounds.ApplyTransform(transform);
+        }
```
[the remark this answers is not in the document]
Comment (zakmat) [reply Gems/Pointcloud/Code/Source/Tools/Components/PointcloudEditorComponent.cpp:84]: good catch. This is trickier than it looks though. There is nothing wrong with the GetLocalBounds - the issue is the result is cached, and before the feature processor is enabled. The proper implementation is to invoke RefreshEntityLocalBoundsUnion on the asset change (probably each reload) which means the FeatureProcessor should signal the event to the controller. This logic is currently non-existent.

[the remark this answers is not in the document]
Comment (norbertprokopiuk) [reply Gems/Pointcloud/Code/Source/Tools/Components/PointcloudEditorComponent.cpp:84]: So my suggestion is to map this as an issue and close and merge this PR. What do you think about this?

[the remark this answers is not in the document]
Comment (zakmat) [reply Gems/Pointcloud/Code/Source/Tools/Components/PointcloudEditorComponent.cpp:84]: I followed the rabbit hole and implemented the ChangeEvent from PointcloudFeatureProcessor to PointcloudController. In the handler itself, I am refreshing cached local bounds which fixes the selection (it seems the selection is a two-step process - the coarse one is using cached bounds to filter entities not visible on the viewport, and the fine one more expensive check from EditorComponentSelectionRequests)

**Review by norbertprokopiuk — APPROVED** [review]
Looks good and works as expected

**Pull request merged**